src/dash-table/DataTable.js

```diff
@@ -178,6 +178,45 @@ export const propTypes = {
             PropTypes.arrayOf(PropTypes.string)
         ]).isRequired,
 
+        number: PropTypes.shape({
+            presentation: PropTypes.oneOf(['input', 'dropdown']),
+            on_change: PropTypes.shape({
+                action: PropTypes.oneOf([
+                    'passthrough',
+                    'coerce',
+                    'validate'
+                ]),
```

Comment:
Nitpicking about nomenclature... maybe:

action: `none`, `coerce`, `validate`
failure: `accept`, `reject`, `default` (clean up the global vs. local scope messiness that comes with skip/reject)

Make the copy/paste all-or-nothing live elsewhere -- e.g. as a table level configuration -- or simply pick one behaviour and out-of-scope the configurable behaviour

Reply:
I like `accept`, `reject`, and `default`. And I agree, all-or-nothing vs cell-by-cell would be cleaner as table-level config. Seems like it could be very confusing if garbage has different effects on the rest of the table depending on which column it's in.

`none` is nicer than `passthrough`, but I still don't see its purpose - how would that be different from `type: 'any'`? On slack you mentioned this would be used if:
> you don’t care that much but maybe you want the initial data to be sane

Is initial data treated differently from changes in some way? Anyway there's also `action: 'validate', failure: 'accept'` which would also never change the value provided, right?

Reply:
So `none` might be interesting and not redundant with `type: any` when taking into consideration wider data type functionality. For example, we want to make the filter type-aware so that entering `> 50` in a number column does not have the same meaning as in a text column. Sorting with type awareness could similarly be impacted. I'm not certain how much of this really requires a `none` or how much of it is a bad idea. We should probably discuss this further.

As currently implemented, initial data isn't handled by the `on_change` prop, neither are programmatically driven changes -- only user-initiated changes are.

Reply:
Filtering and sorting FTW - `action: 'none'` is certainly more intuitive than `action: 'validate', failure: 'accept'` even if the result is equivalent. So OK, let's keep `'none'`.